Added microbenchmarks for profiling, correlating atomic performance on QV100

abhaumick opened the pull request:
- atomic_add_bw_conflict
    - added output dump to validate results
- atomic_add_bw_profile
    - some threads conflict on location 0
    - remaining threads access array[gid]
    - kernel arg ConflictCount for testing
- Updated app makefiles

abhaumick committed:
Microbenchmarks for profiling, correlating atomic performance on QV100

- atomic_add_bw_conflict
    - added output dump to validate results
- atomic_add_bw_profile
    - some threads conflict on location 0
    - remaining threads access array[gid]
    - kernel arg ConflictCount for testing
- Updated app makefiles

FJShen commented on src/cuda/GPU_Microbenchmark/Atomic_ubench/Atomic_add/Atomic_add_bw/atomic_add_bw.cu:
```diff
@@ -84,6 +84,8 @@ int main(){
 	gpuErrchk( cudaMemcpy(startClk, startClk_g, TOTAL_THREADS*sizeof(uint32_t), cudaMemcpyDeviceToHost) );
 	gpuErrchk( cudaMemcpy(stopClk, stopClk_g, TOTAL_THREADS*sizeof(uint32_t), cudaMemcpyDeviceToHost) );
 	gpuErrchk( cudaMemcpy(res, res_g, TOTAL_THREADS*sizeof(int32_t), cudaMemcpyDeviceToHost) );
+	printf("Found GPU Data Value = %d %d %d %d\n", data1[0], data1[1], data1[2], data1[3]);
```
For this and other *.cu files you commited - check if TOTAL_THREADS is greater than 4?

abhaumick replied:
Checking the final values is tricky ...
It changes based on the ConflictCount and number of threads active.
I mostly printed it out to make sense in terms of order of magnitude  and by using a calculator (yeah .. sorry)
I can add more prints but unless someone is really digging into the values .. its not straightforward
But .. can be done I feel .. with some effort

abhaumick committed:
added diverged atomic_add_bw test

- creates fixed diverged warps
    - active threads always conflict on same location
    - inactive threads always inactive

- Updated Makefile

abhaumick commented:
[7f3fd71](https://github.com/accel-sim/gpu-app-collection/pull/11/commits/7f3fd71cdf51cc8efd4e13932f80938c92b362c5) fixes Makefile issue causing Jenkins fail in [67c2a98](https://github.com/accel-sim/gpu-app-collection/pull/11/commits/67c2a98c69aad94c245c18e6a6578c6339884c13)

FJShen commented on src/cuda/GPU_Microbenchmark/Atomic_ubench/Atomic_add/Atomic_add_bw_conflict/atomic_add_bw_conflict_profile.cu:
```diff
@@ -0,0 +1,106 @@
+	asm volatile ("mov.u32 %0, %%clock;" : "=r"(start) :: "memory");
+
+	for (int j=0 ; j<REPEAT_TIMES ; ++j) {
+		atomicAdd(&data1[atomic_loc], 10);
```
Where is atomic_loc declared? I don't see it in this *.cu file.

abhaumick replied:
Corrected this ..
Commented out by mistake

FJShen reviewed (changes requested):
Please remove binary file src/cuda/GPU_Microbenchmark/Atomic_ubench/Atomic_add/Atomic_add_bw_profile/atomic_add_bw_profile away from the repository.

abhaumick committed:
Removed executable atomic_add_bw_profile from repo

abhaumick commented:
> Please remove binary file src/cuda/GPU_Microbenchmark/Atomic_ubench/Atomic_add/Atomic_add_bw_profile/atomic_add_bw_profile away from the repository.

Done.

abhaumick committed:
Restored incorrectly commented out code

- restored atomic_loc selection code in atomic_add_bw_profile
- atomic_loc not initialized otherwise

abhaumick committed:
Delete atomic_add_bw_conflict_profile.cu

removed source file placed in incorrect folder

abhaumick commented:
Should not have made the original commit and PR in a hurry.
Mostly cleaned up .. hope no more stupid mistakes remain.
